### ISSUE TITLE
Using afex for simple one-way ANOVA

### ISSUE THREAD

**Issue by jemus42** [singmann/afex#31]
I was hoping to use `afex` for all my ANOVAing needs, since it's a little easier if I can tell students to just use the afex functions and then the output will match the SPSS results presented in the course I'm tutoring for.

However, unless I misunderstood yet another thing and that's not possible by design, a basic attempt fails:

```
load(url("https://qm.jemu.name/data/ngo")) # The possibly infamous Kähler dataset

# A tibble: 6 × 2
  jahrgang deutsch
     <ord>   <int>
1       11       4
2       11       4
3       11       8
4       11       6
5       11       8
6       11       6

```

```
aov_car(deutsch ~ jahrgang, data = ngo)

Error in if (make.names(name) != name) { : 
  missing value where TRUE/FALSE needed

```
Wheras 
```
aov(deutsch ~ jahrgang * geschl, data = ngo)
```

works just fine.  
Attempting to use `aov_ez` of course did not work, as I do not have an `id` variable in this case, which is a mandatory argument in that function, leading me to assume that `afex` is only meant to work with repeated measure designs or other designs more complex than what I'm used to.

My primary motivation is just to reproduce SPSS results reliably, and give student's an easy way to switch ANOVA types.  

That is also the motiviation between [`tadaatoolbox::tadaa_aov()`](https://github.com/tadaadata/tadaatoolbox/blob/master/R/tadaa_stats.R#L18-L51), but I apparently just pumping models through `car::Anova(…, type = 3)` etc. is not sufficient – but I digress.

TL;DR:

Is a simple one-way ANOVA not possible using afex?

**Comment by singmann** [comment]
The documentation of `aov_car` states (emphasis added):

> The formulas for aov_car or aov_4 **must** contain a single Error term specifying the ID column and potential within-subject factors (you can use mixed for running mixed-effects models with multiple error terms).

So you need to add a `id` variable to your data and also to the call in the error term:
```
ngo$id <- seq_len(nrow(ngo))
aov_car(deutsch ~ jahrgang + Error(id), data = ngo)
```

I am not sure if it is worth the hassle to add the functionality that it runs without it as most real life data sets have a subject identifier column.

I also believe this type of question is more suited for the support forum: http://afex.singmann.science/

**Comment by jemus42** [comment]
Thanks a lot and sorry for not posting in the forum, I checked [the ANOVA subforum but it's apparently empty, so I assumed I had a better chance here](http://afex.singmann.science/forums/forum/anova).

I'm unfamiliar with `Error()`, but now knowing how to use it with an ID column without turning the whole thing into a repeated measure design I can rest easy, thanks again.

**Comment by singmann** [comment]
You could be the one to start populating the forum!

**Comment by jemus42** [comment]
I'm awkward enough as it is being the first one _anywhere_ :<
